### PR TITLE
Collect failure logs recursively

### PR DESCRIPTION
**Pull request by wrobelda**
Fixes microsoft/vcpkg#9802

### CODE REVIEW

**Comment by wrobelda** [comment]
> It seems like you want to recreate the directory tree rather than flattening? Otherwise it seems lots of things getting overwritten is likely.

The `.log` files are being overwritten while copied, anyway, so recreating the structure wouldn't be enough and I'd have to enumerate them: 

```
for (const Path& p : children)
                {
                    filesystem.copy_file(
                        p, target_path / p.filename(), CopyOptions::overwrite_existing, VCPKG_LINE_INFO);
                }
```

However, the artifact files that are we currently copy have unique names assigned by vcpkg (e.g. `autoconf-x64-windows-out.log` or `msys-x64-windows-0-err.log), so my understanding is that any overwriting to happen would be for other log files with more generic names, such us `config.log`, which currently are not copied over at all. 

So the way I see it, is that this fix is a low hanging fruit that will probably cover majority of the cases when additional log files are needed.

**Comment by Neumann-A** [comment]
> So the way I see it, is that this fix is a low hanging fruit that will probably cover majority of the cases when additional log files are needed, without harming the existing functionality.

Won't cover cmake correctly. CMakeOutput.log and CMakeError.log are per subdir.

**Comment by dg0yt** [comment]
I think that this PR isn't the right approach. It is an issue with vcpkg CI, not with vcpkg-tool. In a local build, I can access all logs directly. In fact, when building locally I'm sometimes annoyed by the fact that `config.log` is moved (only) after succesful builds, because I have to look at different locations, depending on build success.

**Comment by strega-nil-ms** [comment]
@dg0yt makes a really good point.

**Comment by BillyONeal** [comment]
I think that most of this infrastructure has been moved into the CI command and out of wrapper scripts suggests that it is now vcpkg-tool's 'fault' now?

File: src/vcpkg/commands.ci.cpp [hunk 1]
```diff
@@ -63,8 +63,11 @@ namespace
             {
                 for (const Path& p : children)
                 {
-                    filesystem.copy_file(
-                        p, target_path / p.filename(), CopyOptions::overwrite_existing, VCPKG_LINE_INFO);
+                    const auto relative_path = StringView(p.parent_path()).substr(source_path.native().size());
```
Comment (BillyONeal) [inline src/vcpkg/commands.ci.cpp:66]:
```suggestion
                    const auto relative_path = StringView(p).substr(source_path.native().size());
```

it seems like the substr 'dominates' any effect of parent_path?

Comment (BillyONeal) [reply src/vcpkg/commands.ci.cpp:66]: Can you update this to use the `proximate` things? https://github.com/microsoft/vcpkg-tool/blob/6334a4f295e8513d4b9c0a391fd761b99278cd0d/include/vcpkg/base/files.h#L240

**Comment by BillyONeal** [comment]
Since this has been draft for 6 months, we're going to close it for now. Please feel free to come back if problems we enumerated above are resolved.

Thanks!

**Pull request closed without merging**